### PR TITLE
Will remove console errors caused by using React.PropTypes and React.createClass

### PR DESCRIPTION
**Pull request by Xyfi**
Will remove console errors caused by using React.PropTypes instead of prop-types package.

Before React would give a deprecation error cause by the use of React.Proptypes instead of the new `prop-types` package. This involved upgrading a few dependencies like `material-ui` and `i18n-calpyso`.

### CODE REVIEW

**Comment by Xyfi** [comment]
Will also fix `React.createClass` error. In the process of fixing it I also updated webpack to version 3+.

File: composites/OnboardingWizard/tests/wizardTest.js [hunk 1]
```diff
@@ -1,9 +1,13 @@
+// console.log( '???' );
```
Comment (thulshof) [inline composites/OnboardingWizard/tests/wizardTest.js]: What's this?

File: composites/OnboardingWizard/tests/wizardTest.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
 jest.unmock( "../OnboardingWizard" );
 jest.unmock( "lodash/cloneDeep" );
 jest.unmock( "../config/production-config" );
 jest.unmock( "material-ui/utils/withWidth" );
 jest.unmock( "../../../utils/i18n" );
-jest.mock( '../helpers/ajaxHelper', () => {
+jest.unmock( "prop-types" );
+// jest.unmock( "[Symbol.iterator]" );
```
Comment (thulshof) [inline composites/OnboardingWizard/tests/wizardTest.js]: Why is this commented?

File: composites/SearchResultForm/SearchResultForm.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from "react";
-import { localize } from "i18n-calypso";
+import PropTypes from "prop-types";
 
+import { localize } from "../../utils/i18n";
```
Comment (thulshof) [inline composites/SearchResultForm/SearchResultForm.js]: Why have you added this? I don't see any new code related to `localize`?

Comment (Xyfi) [inline composites/SearchResultForm/SearchResultForm.js]: I did a search and replace for all occurences of `import { localize } from "i18n-calypso";`, it doesn't seem necessary indeed.

**Comment by thulshof** [comment]
CR :+1:

**Comment by Xyfi** [comment]
I would like to point out that you still get `React.PropTypes` errors in `wordpress-seo`, but these are caused by it's own components.

**Comment by IreneStr** [comment]
Acceptance 👌

**Comment by moorscode** [comment]
@afercia could you create an issue for that ^

**Pull request merged**